pkg/kubelet/status/manager.go

```diff
@@ -131,51 +130,24 @@ func (m *manager) Start() {
 func (m *manager) GetPodStatus(uid types.UID) (api.PodStatus, bool) {
 	m.podStatusesLock.RLock()
 	defer m.podStatusesLock.RUnlock()
-	status, ok := m.podStatuses[uid]
+	status, ok := m.podStatuses[m.podManager.TranslatePodUID(uid)]
 	return status.status, ok
 }
 
 func (m *manager) SetPodStatus(pod *api.Pod, status api.PodStatus) {
-	m.podStatusesLock.Lock()
-	defer m.podStatusesLock.Unlock()
-	oldStatus, found := m.podStatuses[pod.UID]
-
-	// ensure that the start time does not change across updates.
-	if found && oldStatus.status.StartTime != nil {
-		status.StartTime = oldStatus.status.StartTime
-	}
-
-	// Set ReadyCondition.LastTransitionTime.
-	// Note we cannot do this while generating the status since we do not have oldStatus
-	// at that time for mirror pods.
-	if readyCondition := api.GetPodReadyCondition(status); readyCondition != nil {
-		// Need to set LastTransitionTime.
-		lastTransitionTime := unversioned.Now()
-		if found {
-			oldReadyCondition := api.GetPodReadyCondition(oldStatus.status)
-			if oldReadyCondition != nil && readyCondition.Status == oldReadyCondition.Status {
-				lastTransitionTime = oldReadyCondition.LastTransitionTime
-			}
-		}
-		readyCondition.LastTransitionTime = lastTransitionTime
-	}
-
-	// if the status has no start time, we need to set an initial time
-	// TODO(yujuhong): Consider setting StartTime when generating the pod
-	// status instead, which would allow manager to become a simple cache
-	// again.
-	if status.StartTime.IsZero() {
-		if pod.Status.StartTime.IsZero() {
-			// the pod did not have a previously recorded value so set to now
-			now := unversioned.Now()
-			status.StartTime = &now
+	if kubepod.IsMirrorPod(pod) {
```

Comment:
If we are going to write pod status for static pods, I'd suggest change syncPod() in kubelet.go to do that directly.